.gitattributes

```diff
@@ -1,5 +1,14 @@
-* text=auto eol=lf
+* text=auto
+
 Dockerfile* linguist-language=Dockerfile
 vendor.mod linguist-language=Go-Module
 vendor.sum linguist-language=Go-Checksums
-*.bat text eol=crlf
+
+*.go -text diff=golang
```

Comment:
Sorry, I'm still confused :sweat_smile:

Won't this _disable_ Git's line endings enforcement/correction for Go files?  That doesn't seem right?

Also, doesn't `diff=golang` require a `golang` diff tool to be configured? (I don't think Git applies meaning to that by default :sweat_smile:)

Reply:
The original issue was enforcing `lf` line changes for scripts since the build on Windows machines would just fail when their git config was set set to `git config core.autocrlf true`. Since some scripts inside the `scripts/` directory has no file extension, which then lets git treat them as text files converting them to the `crlf` line ending. 

I made a PR to enforce line endings here https://github.com/docker/cli/pull/5216. But it was a bit overzealous and started causing some users to see changes to binary files. https://github.com/docker/cli/issues/5377

This PR pretty much reverts that enforcement for all files except the `scripts` directory.

For `go` I just used this template https://github.com/gitattributes/gitattributes/blob/master/Go.gitattributes. It doesn't require any additional tooling, see https://git-scm.com/docs/gitattributes#_setting_the_internal_diff_algorithm

I tested this branch on Windows and my Linux machine and it seems to be behaving.

Reply:
Ah, I didn't realize `golang` was a built-in diff type, thanks!  I'm still confused by `-text` since `*.go` _is_ text files, but it should be harmless enough.